### ISSUE TITLE
autotools vs cmake: pyilmbase deployment discrepancies

### ISSUE THREAD

**Issue by zachlewis** [AcademySoftwareFoundation/openexr#725]
I could be mistaken, but I *believe* building pyilmbase with autotools produces libraries named 'imathmodule.so', 'iexmodule.so', and 'imathnumpymodule.so'; but building with cmake produces 'imath.so' and 'iex.so' (and, in my case, 'imathnumpy.so' is successfully built, but not installed).

Having built with cmake, other than missing the imathnumpy module, both imath and iex behave just fine for both python-2 and python-3 (on ubuntu-18.04 and centos-7.4; on osx-10.14.6, python-3 behaves, but my python-2 builds segfault -- I'm assuming that's a problem on my end, and I'll open a separate ticket if the issue persists).

That said, it seems that external libraries built against PyIlmBase  are expecting to find 'imathmodule' (PyAlembic in particular; various others, according to my brief search on github). I could see this becoming a larger problem the longer the discrepancy exists.

**Comment by maxnbk** [comment]
I am experiencing this issue (lacking module suffix on the libs when using cmake), and happy to fix it in my own build chain, but would like to understand which is meant to be correct. the module suffix naming always seemed weird, but the now baked-in expectation in dependencies of one name or the other, and the discrepancy between the two build methods seems like an urgent thing to resolve.

**Comment by cary-ilm** [comment]
This did apparently change with the rewrite of the cmake configuration files. @kdt3rd pointed out that python searches first for imath.so and then for imathmodule.so if that fails, so we eliminate a stat() call by eliminating the "module".   

We can easily fix the autotools build to be consistent, but @maxnbk, you mention baked-in expectations for the module name. How baked-in? 

What is the community preference here? 
1. revert the cmake config to build imathmodule.so to preserve old behavior?
2. update the autotools build to use the abbreviated imath.so.

**Comment by maxnbk** [comment]
Mainly just every version of Alembic ever, not that that isn't trivially fixable:
https://github.com/alembic/alembic/blob/3c48ce60596c39735a1e923c0c0a5f78a17225c0/cmake/Modules/FindPyIlmBase.cmake#L118
Easily patchable too. if imath.so is preferred, it sounds to me like fixing the autoconf one is the way to go and cmake is doing it "correctly", so we'll move to that - Since I'm using cmake anyway, that is fine for me, but still good to address globally to avoid confusion down the line.

**Comment by zachlewis** [comment]
As long as there's consistency, I don't really have any preference for whether we go abbreviated or old-style. I had thought earlier that there were a bunch of projects on github expecting 'imathmodule', but upon closer inspection, everything I can find really is either an Alembic fork or an Alembic plugin of some sort; so I guess I'll draw their attention to this thread.


The only remaining inconsistencies I'm seeing are regarding 'imathnumpy' -- but now I'm not sure that what I'm seeing is a difference between cmake and autotools, or a difference between building the full shebang (openexr + ilmbase + pyilmbase) at once, vs building pyilmbase separately. I can do some comparative testing later this week, and I can move the below to a separate issue if it would be more appropriate.

What I can say is, for cmake + the full shebang, on Ubuntu-18, Centos-7, and OSX-10.14, what I'm finding is:
  - imathnumpy.so does seem compile and link just fine, even though it doesn't get installed when making the full shebang (openexr+ilmbase+pyilmbase); but manually copying imathnumpy.so from the build directory to the install directory afterwards seems to work...
  - Although (or because?) I'm using cmake, I think something is going a little wonky with the paths in some of the generated pkg-config files -- but I'm not pkg-config-savvy enough to know for sure if what I'm seeing is problematic. 

PyIlmBase.pc (uses relative paths for exec_prefix, includedir, and PyIlmBase_includedir -- is that allowed?):
```
...
prefix=/packages/openexr/2.5.1
exec_prefix=bin
libdir=/packages/openexr/2.5.1/lib
includedir=include
PyIlmBase_includedir=include/OpenEXR
...
```

...versus IlmBase.pc (exec_prefix should be ${prefix}/bin?):
```
...
prefix=/packages/openexr/2.5.1
exec_prefix=${prefix}
libdir=/packages/openexr/2.5.1/lib
includedir=${prefix}/include
libsuffix=-2_5
...
```

...versus OpenExr.pc (libdir is prefixed with extra '${exec_prefix}/'):
```
...
prefix=/packages/openexr/2.5.1
exec_prefix=${prefix}
libdir=${exec_prefix}//packages/openexr/2.5.1/lib
includedir=${prefix}/include
OpenEXR_includedir=${prefix}/include/OpenEXR
libsuffix=-2_5
...
```

**Comment by cary-ilm** [comment]
fixed by #746
